test/tls_record.cpp

```diff
@@ -77,8 +84,78 @@ tls_handshake::message_type read_message(tls_handshake::handshake_type t, net::c
   WINTLS_UNREACHABLE_RETURN(0);
 }
 
+tls_extension::message_type read_message(tls_extension::extension_type t,
+                                         net::const_buffer& buffer,
+                                         std::uint16_t size) {
+  switch (t) {
+    case tls_extension::extension_type::server_name:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::max_fragment_length:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::status_request:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::supported_group:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::signature_algorithms:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::use_srtp:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::heartbeat:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::application_layer_protocol_negotiation:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::signed_certificate_timestamp:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::client_certificate_type:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::server_certificate_type:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::padding:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::pre_shared_key:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::early_data:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::supported_versions:
+      return tls_extension::supported_versions{buffer};
+    case tls_extension::extension_type::cookie:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::psk_key_exchange_modes:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::certificate_authorities:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::oid_filters:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::post_handshake_auth:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::signature_algorithms_cert:
+      return tls_extension::common{buffer, size};
+    case tls_extension::extension_type::key_share:
+      return tls_extension::common{buffer, size};
+    default:
+      return tls_extension::common{buffer, size};
+  }
```

Comment:
I might be reading this the wrong way, but it definitely looks like this could more or less be reduced to:

`return tls_extension::common{buffer, size};`

Also, if you are going to switch on all the enum values, then please leave out the default case. That way the compiler can warn if an enum value isn't handled which is definitely nice.

So maybe this could be written as:

```
 switch (t) {
    case tls_extension::extension_type::server_name:
    case tls_extension::extension_type::max_fragment_length:
    case tls_extension::extension_type::status_request:
    // Remaing enum values follow...
     return tls_extension::common{buffer, size};
 }
WINTLS_UNREACHABLE_RETURN(0);
```